torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -642,6 +642,208 @@ class IrParser {
           });
     }
 
+    {
+      auto ptr_op = getOperatorForLiteral(
+          "aten::native_layer_norm(Tensor input, Tensor? weight, Tensor? bias, int M, int N, float eps) -> (Tensor, Tensor, Tensor)");
+      registerParseRule(
+          ptr_op,
+          [](const Node* node,
+             std::unordered_map<size_t, CgValue>& value_map) -> void {
+            auto input = value_map[node->input(0)->unique()]->as<TensorView>();
+
+            TensorView* weight = nullptr;
+            if (!node->input(1)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              weight = value_map[node->input(1)->unique()]->as<TensorView>();
+            }
+
+            TensorView* bias = nullptr;
+            if (!node->input(2)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              bias = value_map[node->input(2)->unique()]->as<TensorView>();
+            }
+
+            // M = product of [0, reduction_axis)
+            auto M = constant_as<int>(node->input(3));
```

Comment:
Mentioned this in the comment down in `symbolic_script.cpp`, it's unrealistic to expect `M/N` to be constant here without the proper conversion/guard.